feat: Document how to use SQLite3 and remove PostgreSQL-only mentions (#524)
serverpod/serverpod_docs: docs/06-concepts/06-database/01-connection.md

# Connection

In Serverpod the connection details and password for the database are stored inside the `config` directory in your server package. Serverpod automatically establishes a connection to the database instance by using these configuration details when you start the server.

> If using PostgreSQL, the easiest way to get started is to use a Docker container to run your local PostgreSQL server, and this is how Serverpod is set up out of the box. This page contains more detailed information if you want to connect to another database instance or run PostgreSQL locally yourself.

Reviewer: The rest of this page uses `Postgres`. Worth picking one and using it consistently.

Author: Done at https://github.com/serverpod/serverpod_docs/pull/524/commits/fbc4af43e141624ebd49157c2f3b4cbe02c4818e.

You can also configure this setting via the environment variable `SERVERPOD_DATABASE_MAX_CONNECTION_COUNT`.

> On SQLite, this configuration will set the number of read-only transactions that can be executed concurrently - only one write transaction can be executed at a time.

Reviewer: Removing the em dash and making the text clearer.

```suggestion
On SQLite, this configuration sets the number of read-only transactions that can run concurrently. Only one write transaction can run at a time.
```

Author: Fixed at https://github.com/serverpod/serverpod_docs/pull/524/commits/fbc4af43e141624ebd49157c2f3b4cbe02c4818e.
